providers/digitalocean2/src/main/java/org/jclouds/digitalocean2/compute/DigitalOcean2ComputeServiceAdapter.java

```diff
@@ -96,16 +106,15 @@
          options.addSshKeyIds(templateOptions.getSshKeyIds());
       }
 
-      Map<String, String> metadataAndTags = metadataAndTagsAsCommaDelimitedValue(templateOptions);
-      if (!metadataAndTags.isEmpty()) {
-         @SuppressWarnings("unchecked")
-         List<String> regionFeatures = (List<String>) template.getLocation().getMetadata().get("features");
-         if (regionFeatures.contains("metadata")) {
-            options.userData(json.toJson(metadataAndTags));
-         } else {
-            logger.debug(">> region %s does not support metadata, ignoring provided user data", template.getLocation()
-                  .getId());
-         }
+      // In DigitalOcean, user_data is a SINGLE string, NOT a map!
+      // Encoding tags or anything else than user_data in here breaks their functionality.
+      if (null != templateOptions.getUserData()) {
+         setUserDataIfSupported(template, options, new String(templateOptions.getUserData()));
+      }
+      // Backwards compatible variant, getting userData from userMetaData map.
+      Map<String, String> metadataAndTags = templateOptions.getUserMetadata();
+      if (null != metadataAndTags.get("user_data")) {
+         setUserDataIfSupported(template, options, metadataAndTags.get("user_data"));
       }
```

Comment:
Set this in an `else` block for the first conditional to only call the backwards compatible logic if the new method has not been configured?